Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabaseTransaction.cpp

```diff
@@ -77,21 +81,31 @@ void UniqueIDBDatabaseTransaction::abort()
 {
     LOG(IndexedDB, "UniqueIDBDatabaseTransaction::abort");
     
-    auto database = databaseConnection().database();
-    ASSERT(database);
+    if (!database())
+        return;
 
-    database->abortTransaction(*this, [this](auto& error) {
+    database()->abortTransaction(*this, [this, weakThis = WeakPtr { *this }](auto& error) {
         LOG(IndexedDB, "UniqueIDBDatabaseTransaction::abort (callback)");
 
-        databaseConnection().didAbortTransaction(*this, error);
+        if (weakThis && m_databaseConnection)
+            m_databaseConnection->didAbortTransaction(*this, error);
     });
 }
 
 void UniqueIDBDatabaseTransaction::abortWithoutCallback()
 {
     LOG(IndexedDB, "UniqueIDBDatabaseTransaction::abortWithoutCallback");
 
-    databaseConnection().abortTransactionWithoutCallback(*this);
+    if (m_databaseConnection)
+        m_databaseConnection->abortTransactionWithoutCallback(*this);
+}
+
+UniqueIDBDatabase* UniqueIDBDatabaseTransaction::database() const
+{
+    if (!m_databaseConnection)
+        return nullptr;
+
+    return m_databaseConnection->database();
```

Comment:
Could be a one line:
return m_databaseConnection ? m_databaseConnection->database() : nullptr;

Reply:
Sure